src/aap_eda/core/models/activation.py

```diff
@@ -93,6 +139,44 @@ class Meta:
     activation_pod_id = models.TextField(null=True)
     status_message = models.TextField(null=True, default=None)
 
+    def save(self, *args, **kwargs):
+        # when creating
+        if self._state.adding:
+            if self.status_message is None:
+                self.status_message = ACTIVATION_STATUS_MESSAGE_MAP[
+                    self.status
+                ]
+        else:
+            if not bool(kwargs):
+                raise ValidationError(
+                    _(
+                        "update_fields is required to use when saving "
+                        "due to race conditions"
+                    )
+                )
+            # when updating without status
+            elif (
+                "status_message" in kwargs["update_fields"]
+                and "status" not in kwargs["update_fields"]
+            ):
+                raise ValidationError(
+                    _(
+                        "status_message is required to update with "
+                        "status together"
+                    )
+                )
+            # when updating without status_message
+            elif (
+                "status" in kwargs["update_fields"]
+                and "status_message" not in kwargs["update_fields"]
+            ):
+                self.status_message = ACTIVATION_STATUS_MESSAGE_MAP[
+                    self.status
+                ]
+                kwargs["update_fields"].append("status_message")
+
+        super().save(*args, **kwargs)
```

Comment:
@Alex-Izquierdo @hsong-rh 
Would it make sense to wrap this in an select_for_update or should that be done by the caller?

Reply:
I would leave it to the caller, let's see how this restriction works for now.